src/components/app/details/triggerView/PipelineConfigDiff/usePipelineDeploymentConfig.ts

```diff
@@ -117,47 +121,39 @@ export const usePipelineDeploymentConfig = ({
     const specificDeploymentConfig = pipelineDeploymentConfigRes?.[2]?.result ?? null
     const isLastDeployedConfigAvailable = pipelineDeploymentConfigRes?.[0] !== null
 
-    const [deploymentTemplateResolvedDataLoader, deploymentTemplateResolvedData, deploymentTemplateResolvedDataErr] =
-        useAsync(
-            () =>
-                abortPreviousRequests(() => {
-                    const compareData = getComparisonDataBasedOnDeploy({
-                        deploy,
-                        latestDeploymentConfig,
-                        specificDeploymentConfig,
-                        recentDeploymentConfig,
-                    })
-
-                    return Promise.all([
-                        getAppEnvDeploymentConfig({
-                            params: {
-                                configArea: 'ResolveData',
-                                appName,
-                                envName,
-                            },
-                            payload: {
-                                values: getDeploymentTemplateValues(compareData?.deploymentTemplate),
-                            },
-                            signal: deploymentTemplateResolvedDataAbortControllerRef.current?.signal,
-                        }),
-                        recentDeploymentConfig
-                            ? getAppEnvDeploymentConfig({
-                                  params: {
-                                      configArea: 'ResolveData',
-                                      appName,
-                                      envName,
-                                  },
-                                  payload: {
-                                      values: getDeploymentTemplateValues(recentDeploymentConfig?.deploymentTemplate),
-                                  },
-                                  signal: deploymentTemplateResolvedDataAbortControllerRef.current?.signal,
-                              })
-                            : { result: null },
-                    ])
-                }, deploymentTemplateResolvedDataAbortControllerRef),
-            [convertVariables, pipelineDeploymentConfigRes],
-            convertVariables && !!pipelineDeploymentConfigRes,
-        )
+    const [
+        deploymentTemplateResolvedDataLoader,
+        deploymentTemplateResolvedData,
+        deploymentTemplateResolvedDataErr,
+        reloadDeploymentTemplateResolvedData,
+    ] = useAsync(
+        () =>
+            abortPreviousRequests(() => {
+                const compareData = getComparisonDataBasedOnDeploy({
+                    deploy,
+                    latestDeploymentConfig,
+                    specificDeploymentConfig,
+                    recentDeploymentConfig,
+                })
+
+                return Promise.all([
+                    getDeploymentTemplateResolvedData({
+                        appName,
+                        envName,
```

Comment:
We write separate service for such cases in general. Pls confirm with @vivek-devtron

Reply:
discussed not required in this case